### PR TITLE
KT-33741 assertNotNull(deferred) gets "Deferred result is never used" inspection

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-33741

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/coroutines/DeferredResultUnusedInspection.kt [hunk 1]
```diff
@@ -22,10 +23,12 @@ class DeferredResultUnusedInspection(@JvmField var standardOnly: Boolean = false
                 (!inspection.standardOnly || expression.calleeExpression?.text in shortNames),
     callChecker = fun(resolvedCall, inspection): Boolean {
         if (inspection !is DeferredResultUnusedInspection) return false
+        val resultingDescriptor = resolvedCall.resultingDescriptor
+        if (resultingDescriptor.fqNameSafe == FqName("kotlin.test.assertNotNull")) return false
```
Comment (abelkov) [inline idea/src/org/jetbrains/kotlin/idea/inspections/coroutines/DeferredResultUnusedInspection.kt]: But the called function may have an arbitrary name, see https://youtrack.jetbrains.com/issue/KT-33741#focus=streamItem-27-3697460.0-0

**Comment by nikita-nazarov** [comment]
Hello @t-kameyama. It's been a long time since you uploaded this pr. But it can't be merged in the current state. As @abelkov mentioned, the problem with functions with arbitrary names remains unsolved.

**Comment by t-kameyama** [comment]
@nikita-nazarov Fixed, PTAL.

**Comment by t-kameyama** [comment]
I've created new pull request in intellij-community, please review https://github.com/JetBrains/intellij-community/pull/1605.

**Pull request closed without merging**